### PR TITLE
fix todo:pass the cluster role from env and add err check in e2e

### PR DESCRIPTION
**Pull request by hanxiaoshuai**
fix todo:pass the cluster role from env and add err check in e2e

### CODE REVIEW

**Comment by shivramsrivastava** [comment]
@hanxiaoshuai There is a conflict please rebase and fix it.

**Comment by hanxiaoshuai** [comment]
@shivramsrivastava I have rebased and fixed it, Thanks

**Comment by shivramsrivastava** [comment]
Thanks

/lgtm

File: test/e2e/framework/deployment_util.go [hunk 1]
```diff
@@ -100,52 +101,52 @@ func (f *Framework) DeleteDeploymentIfExist(nsName string, deploymentName string
 			return nil
 		} else {
 			// error occurred while trying to fetch the deployment info
-			Logf("Unable to fetch the deployment %v", deploymentName)
-			return err
+			return fmt.Errorf("unable to fetch the deployment %v. error:%v", deploymentName, err)
```
Comment (m1093782566) [inline test/e2e/framework/deployment_util.go]: nit: `error: %v`, a whitespace please.

File: test/e2e/framework/namespace.go [hunk 2]
```diff
@@ -55,15 +57,13 @@ func (f *Framework) deleteNamespaceIfExist(nsName string) error {
 			Logf("%v dosent not exist, no need to delete non existing namespace", nsName)
 			return nil
 		} else {
-			Logf("error occurred while fetching %v for deleting", nsName)
-			return err
+			return fmt.Errorf("error occurred while fetching %v for deleting:%v", nsName, err)
```
Comment (m1093782566) [inline test/e2e/framework/namespace.go]: `deleting: %v`

File: test/e2e/framework/deployment_util.go [hunk 3]
```diff
@@ -100,52 +101,52 @@ func (f *Framework) DeleteDeploymentIfExist(nsName string, deploymentName string
...
-	err := f.ClientSet.AppsV1beta1().Deployments(nsName).Delete(deploymentName, &metav1.DeleteOptions{})
-	if errors.IsNotFound(err) {
-		Logf("%v deployment doesn't exist", deploymentName)
-		return nil
-	} else {
-		Logf("Unable to delete the deployment %v from namespace %v", deploymentName, nsName)
-		return err
+	if err := f.ClientSet.AppsV1beta1().Deployments(nsName).Delete(deploymentName, &metav1.DeleteOptions{}); err != nil {
+		if errors.IsNotFound(err) {
+			Logf("%v deployment doesn't exist", deploymentName)
```
Comment (m1093782566) [inline test/e2e/framework/deployment_util.go]: nit: Logf("Deployment %v doesn't exist", deploymentName)

File: test/e2e/framework/framework.go [hunk 4]
```diff
@@ -107,18 +108,32 @@ func (f *Framework) BeforeEach() {
 
 	}
 
-	Logf("Posedion test are pointing to %v", *kubeConfig)
+	Logf("Poseidon test are pointing to %v", *kubeConfig)
 
-	_ = f.DeleteService(f.TestingNS, "poseidon")
-	_ = f.DeleteService(f.TestingNS, "firmament-service")
+	if err := f.DeleteService(f.TestingNS, "poseidon"); err != nil {
+		Logf("error deleting service poseidon: %v", err)
```
Comment (m1093782566) [inline test/e2e/framework/framework.go]: nit: Logf("Error deleting service poseidon: %v", err)

File: test/e2e/framework/framework.go [hunk 5]
```diff
@@ -107,18 +108,32 @@ func (f *Framework) BeforeEach() {
...
+	}
+	if err := f.DeleteService(f.TestingNS, "firmament-service"); err != nil {
+		Logf("error deleting service firmament-service: %v", err)
```
Comment (m1093782566) [inline test/e2e/framework/framework.go]: nit: Logf("Error deleting service firmament-service: %v", err)

File: test/e2e/framework/framework.go [hunk 6]
```diff
@@ -107,18 +108,32 @@ func (f *Framework) BeforeEach() {
...
+	}
 
-	// TODO(shiv): We need to pass the cluster role from env
-	_ = f.DeletePoseidonClusterRole("poseidon", f.TestingNS)
+	if len(*clusterRole) == 0 {
+		*clusterRole = "poseidon"
+	}
+	if err := f.DeletePoseidonClusterRole(*clusterRole, f.TestingNS); err != nil {
+		Logf("error deleting cluster role: %v", err)
```
Comment (m1093782566) [inline test/e2e/framework/framework.go]: same as above - log message should be started with an upper-case word.

File: test/e2e/framework/framework.go [hunk 7]
```diff
@@ -107,18 +108,32 @@ func (f *Framework) BeforeEach() {
...
+	}
 
 	// This is needed if we have a dirty test run which leaves the pods and deployments hanging
-	_ = f.deleteNamespaceIfExist(f.TestingNS)
-	_ = f.DeleteDeploymentIfExist(f.TestingNS, poseidonDeploymentName)
-	_ = f.DeleteDeploymentIfExist(f.TestingNS, firmamentDeploymentName)
+	if err := f.deleteNamespaceIfExist(f.TestingNS); err != nil {
+		Logf("error occurred when delete namespace if exist: %v", err)
```
Comment (m1093782566) [inline test/e2e/framework/framework.go]: same as above.

File: test/e2e/framework/framework.go [hunk 8]
```diff
@@ -107,18 +108,32 @@ func (f *Framework) BeforeEach() {
...
+	}
+	if err := f.DeleteDeploymentIfExist(f.TestingNS, poseidonDeploymentName); err != nil {
+		Logf("error occurred when delete deployment if exist: %v", err)
```
Comment (m1093782566) [inline test/e2e/framework/framework.go]: same as above.

File: test/e2e/framework/framework.go [hunk 9]
```diff
@@ -107,18 +108,32 @@ func (f *Framework) BeforeEach() {
...
+	}
+	if err := f.DeleteDeploymentIfExist(f.TestingNS, firmamentDeploymentName); err != nil {
+		Logf("error occurred when delete deployment if exist: %v", err)
```
Comment (m1093782566) [inline test/e2e/framework/framework.go]: same as above.

**Comment by m1093782566** [comment]
Some more nit picking, otherwise LGTM.

**Comment by hanxiaoshuai** [comment]
@m1093782566 Done

**Pull request merged**